qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -409,17 +409,64 @@ public static void waitForElasticsearch(String status, String index, Installatio
         withLogging(() -> ServerUtils.waitForElasticsearch(status, index, installation, username, password));
     }
 
-    private static void withLogging(ThrowingRunnable r) throws Exception {
+    /**
+     * Runs the provided closure, and captures logging information if an exception is thrown.
+     * @param r the closure to run
+     * @throws Exception any exception encountered while running the closure are propagated.
+     */
+    private static <E extends Exception> void withLogging(CheckedRunnable<E> r) throws Exception {
         try {
             r.run();
         } catch (Exception e) {
-            final Shell.Result logs = sh.run("docker logs " + containerId);
+            final Shell.Result logs = getContainerLogs();
             logger.warn("Elasticsearch container failed to start.\n\nStdout:\n" + logs.stdout + "\n\nStderr:\n" + logs.stderr);
             throw e;
         }
     }
 
-    private interface ThrowingRunnable {
-        void run() throws Exception;
+    public static JsonNode getJson(String path) throws IOException {
+        final String pluginsResponse = makeRequest(Request.Get("http://localhost:9200/" + path));
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        return mapper.readTree(pluginsResponse);
+    }
+
+    public static List<String> getEnabledPlugins(Installation installation) throws Exception {
```

Comment:
Why is this something docker specific? Seems applicable to all distros, so should be in ServerUtils?